### PR TITLE
Switch MongoDB libraries

### LINKED ISSUE #5795: MongoDB plugin SSL connects but fails to authenticate
### Relevant telegraf.conf:

```toml
[[inputs.mongodb]]
  servers = ["mongodb://host-1.datacenter.project.domain.com:27017,host-2.datacenter.project.domain.com:27017,host-3.datacenter.project.domain.com:27017/replicaSet=my-replSet&ssl=true&authMechanism=MONGODB-X509&authSource=$external"]

  ## When true, collect per database stats
  gather_perdb_stats = true

  ## Optional TLS Config
  ssl_ca = "/etc/telegraf/ca.pem"
  ssl_cert = "/etc/telegraf/cert.pem"
  ssl_key = "/etc/telegraf/key.pem"

  ## Use TLS but skip chain & host verification
  # insecure_skip_verify = false
```

### System info:

Telegraf Version: Telegraf 1.10.3 (git: HEAD 294bb666)
MongoDB Version: 4.0.9
OS: Ubuntu 16.06 x86_64
SSL
### Steps to reproduce:

1. User is created in $external db for external x509 authentication
2. Start telegraf service with above configuration

### Expected behavior:
The plugin should feed the output source (in my case influxdb) with metrics relating to the databases. 

### Actual behavior:

Error generated in log.
I'm able to log into the database using the same certificates provided to telegraf remotely with the mongo CLI and I'm able to successfully run the db.serverStatus() command

#### Telegraf Log
```
2019-05-02T16:40:30Z E! [inputs.mongodb]: Error in plugin: command serverStatus requires authentication
```
#### MongoDB auth info

```
{
	"_id" : "$external.CN=telegraf-monitor,OU=dev-mongo,O=Acme International Corp.,L= Arrakeen,ST= Arrakis,C=DN",
	"user" : "CN=telegraf-monitor,OU=dev-mongo,O=Acme International Corp.,L= Arrakeen,ST= Arrakis,C=DN",
	"db" : "$external",
	"roles" : [
		{
			"role" : "clusterMonitor",
			"db" : "admin"
		}
	],
	"mechanisms" : [
		"external"
	]
}
```

### LINKED ISSUE #7706: Update mongodb input to use maintained library
As far as I could see, the MongoDB input plugin uses `gopkg.in/mgo.v2` as Mongo Driver, which seems to be unmaintained: https://github.com/go-mgo/mgo/blob/v2-unstable/README.md

There exists a MongoDB Go Driver directly from MongoDB: https://docs.mongodb.com/drivers/go

Is there any reason, why the plugin still uses the old unmaintained driver?

### PR DESCRIPTION
**Pull request by akrantz01**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

resolves #7706 
resolves #5795

Switched to the [official MongoDB driver](https://github.com/mongodb/mongo-go-driver) from [mgo v2](https://github.com/go-mgo/mgo/tree/v2) that has been unmaintained for ~4 years. Certificate-based authentication, SRV domains in connection URIs using `mongodb+srv://`, and all query parameters should now be supported.

### CODE REVIEW

**Comment by sspaink** [comment]
Awesome work @akrantz01! Looks like changing the library was definitely necessary. It does look like like the `Init()` method isn't called by any of the unit tests, I assume you tested this locally? Do you think it is possible to add a test that covers it perhaps?

**Comment by akrantz01** [comment]
I don't think there's a way to add a unit test covering it, but I changed the integration tests so that it gets used.

**Review by sspaink — APPROVED** [review]
lgtm! Thank you for the changes.

File: plugins/inputs/mongodb/mongodb.go [hunk 1]
```diff
@@ -78,118 +81,104 @@ func (*MongoDB) Description() string {
...
+		if len(m.Ssl.CaCerts) > 0 {
+			roots := x509.NewCertPool()
+			for _, caCert := range m.Ssl.CaCerts {
+				if ok := roots.AppendCertsFromPEM([]byte(caCert)); !ok {
+					return fmt.Errorf("failed to parse root certificate")
+				}
+			}
+			tlsConfig.RootCAs = roots
+		} else {
+			tlsConfig.InsecureSkipVerify = true
```
Comment (ssoroka) [inline plugins/inputs/mongodb/mongodb.go:98]: I see this isn't new code, but I'm not a fan of subtly letting the user enable SSL but then quietly skipping certs and connecting, as it's vulnerable to MITM (man in the middle) attacks. I think we've explicitly made the user set `insecure_skip_verify = true` in other places.

File: plugins/inputs/mongodb/mongodb.go [hunk 2]
```diff
@@ -78,118 +81,104 @@ func (*MongoDB) Description() string {
...
-			if err != nil {
-				m.Log.Errorf("Error in plugin: %v", err)
-			}
-		}(m.getMongoServer(u))
-	}
-
-	wg.Wait()
-	return nil
-}
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (ssoroka) [inline plugins/inputs/mongodb/mongodb.go:125]: just defer cancel() here?

**Review by ssoroka — APPROVED** [review]
approved with comments you can consider addressing

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**